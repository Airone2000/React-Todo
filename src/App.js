import React from 'react';
import MainMenu from './MainMenu';
import {BrowserRouter as Router, Route} from 'react-router-dom';
import {default as TodoContainer} from './TodoList/TodoListContainer';
import About from './TodoList/About';
import './index.css';

export default function App() {
  return (
    <Router>
      <React.Fragment>
        <MainMenu />
        <Route path="/" exact={true} component={TodoContainer} />
        <Route path="/about" component={About} />
      </React.Fragment>
    </Router>
  );
}