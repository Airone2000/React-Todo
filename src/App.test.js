import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import renderer from "react-test-renderer";

function Link({ href, children }) {
  return <a href={href}>{children}</a>;
}

it("renders without crashing", () => {
  const snapshot = renderer
    .create(<Link href="index.html">Mon bouton</Link>)
    .toJSON();
  console.log(snapshot);
  expect(snapshot).toMatchSnapshot();
});

it("renders once again but with google as href", () => {
  const snapshot = renderer
    .create(<Link href="google.us" children="GOOGLE US" />)
    .toJSON();
  expect(snapshot).toMatchSnapshot();
});

it("will test inlineSnapshot", () => {
  const tree = renderer
    .create(<Link href="google.us" children="GOOGLE USS" />)
    .toJSON();

  expect(tree).toMatchInlineSnapshot(`
<a
  href="google.us"
>
  GOOGLE USS
</a>
`);
});


it('will test object', () => {
  const person = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'Florence'
  };

  expect(person).toMatchSnapshot({
    createdAt: expect.any(Date),
    id: expect.any(Number),
    name: expect.stringMatching('Florence')
  });
});
