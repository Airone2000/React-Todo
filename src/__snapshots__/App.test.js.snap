// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders once again but with google as href 1`] = `
<a
  href="google.us"
>
  GOOGLE US
</a>
`;

exports[`renders without crashing 1`] = `
<a
  href="index.html"
>
  Mon bouton
</a>
`;

exports[`will test object 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": StringMatching /Florence/,
}
`;
